addons/binding/org.openhab.binding.elerotransmitterstick/src/main/java/org/openhab/binding/elerotransmitterstick/stick/TransmitterStick.java

```diff
@@ -0,0 +1,402 @@
+            }
+        }
+
+        return true;
+    }
+
+    static class DueCommandSet extends TreeSet<Command> {
+        private static final long serialVersionUID = -3216360253151368826L;
+
+        public DueCommandSet() {
+            super(new Comparator<Command>() {
+                /**
+                 * Due commands are sorted by priority first and then by delay.
+                 */
+                @Override
+                public int compare(Command o1, Command o2) {
+                    if (o1.equals(o2)) {
+                        return 0;
+                    }
+
+                    int d = o2.getPriority() - o1.getPriority();
+                    if (d < 0) {
+                        return -1;
+                    }
+
+                    if (d == 0 && o1.getDelay(TimeUnit.MILLISECONDS) < o2.getDelay(TimeUnit.MILLISECONDS)) {
+                        return -1;
+                    }
+                    return 1;
+                }
+            });
+        }
+
+        @Override
+        public boolean add(Command e) {
+            if (TransmitterStick.prepareAddition(e, this)) {
+                return super.add(e);
+            }
+
+            return false;
+        }
+    }
+
+    class CommandWorker implements Runnable {
+        private ArrayList<Integer> validIds = new ArrayList<>();
+        private final AtomicBoolean terminated = new AtomicBoolean();
+        private final int updateInterval;
+        private final SerialConnection connection;
+
+        private final BlockingQueue<Command> cmdQueue = new DelayQueue<Command>() {
+            @Override
+            public boolean add(Command e) {
+                if (TransmitterStick.prepareAddition(e, this)) {
+                    return super.add(e);
+                }
+
+                return false;
+            }
+        };
+
+        CommandWorker() {
+            connection = new SerialConnection(config.portName);
+            updateInterval = config.updateInterval;
+        }
+
+        void terminateUpdates() {
+            terminated.set(true);
+
+            // add a NONE command to make the thread exit from the call to take()
+            cmdQueue.add(new Command(CommandType.NONE));
+        }
+
+        void requestUpdate(Integer... channelIds) {
+            // this is a workaround for a bug in the stick firmware that does not
+            // handle commands that are sent to multiple channels correctly
+            if (channelIds.length > 1) {
+                for (int channelId : channelIds) {
+                    requestUpdate(channelId);
+                }
+            } else if (channelIds.length == 1) {
+                logger.debug("adding INFO command for channel id {} to queue...", Arrays.toString(channelIds));
+                cmdQueue.add(new DelayedCommand(CommandType.INFO, 0, Command.FAST_INFO_PRIORITY, channelIds));
+            }
+        }
+
+        void executeCommand(CommandType command, Integer... channelIds) {
```

Comment:
Wouldn't it be better to use `void executeCommand(CommandType command, List<Integer> channelIds)`? because then the callers can pass their `ArrayList`s directly.